File: 2022/go/01.go
package main

import (
	"fmt"
	"os"
	"sort"
	"strconv"
	"strings"
)

func main() {
	input, _ := os.ReadFile("2022/01.input")
	chunks := strings.Split(strings.Trim(string(input), "\n"), "\n\n")

	calories := make([]int, len(chunks))
	for i, lines := range chunks {
		for _, line := range strings.Split(lines, "\n") {
			calorie, _ := strconv.Atoi(line)
			calories[i] += calorie
		}
	}

	sort.Sort(sort.Reverse(sort.IntSlice(calories)))

	first := calories[0]
	fmt.Println(first) // 68775

	second := calories[0] + calories[1] + calories[2]
	fmt.Println(second) // 202585
}

File: 2022/typescript/01.ts
import {readFileSync} from 'fs'

const input: number[][] = readFileSync('./2022/01.input', 'utf8')
  .split(/\n\n/)
  .map(chunk => chunk.split(/\n/).map(line => Number(line)))

const calories: number[] = input.map(a => a.reduce((a, e) => a + e)).sort((a, b) => b - a)

const first = calories[0]
console.log(first)

const second = calories.slice(0, 3).reduce((a, e) => a + e)
console.log(second)
